src/strategyHelpers/blueGreen/deploy.ts

```diff
@@ -56,21 +65,35 @@ export async function deployBlueGreenSMI(
       manifestObjects.unroutedServiceEntityList
    )
 
-   await deployObjects(kubectl, newObjectsList)
+   const otherObjDeployment: DeployResult = await deployObjects(
+      kubectl,
+      newObjectsList
+   )
 
    // make extraservices and trafficsplit
-   await setupSMI(kubectl, manifestObjects.serviceEntityList)
+   const smiAndSvcDeployment = await setupSMI(
+      kubectl,
+      manifestObjects.serviceEntityList
+   )
 
    // create new deloyments
    const blueGreenDeployment: BlueGreenDeployment = await deployWithLabel(
       kubectl,
       manifestObjects.deploymentEntityList,
       GREEN_LABEL_VALUE
    )
-   return {
-      deployResult: blueGreenDeployment.deployResult,
-      objects: [].concat(blueGreenDeployment.objects, newObjectsList)
-   }
+
+   blueGreenDeployment.objects.push(...newObjectsList)
+   blueGreenDeployment.objects.push(...smiAndSvcDeployment.objects)
```

Comment:
not very clear from just the names what the difference between objects and deploy result is. Is there a good name we could change objects to?

Reply:
the deploy result stores JSON objects of the k8s objects it deploys as well as the manifest files that those objects come from, so figured that would all be contained within one result coming back from when we deploy something. I could change it to k8sobjects if you think that's better but feel like that's implied?

Reply:
Seems fine for now!